<!-- -*- mode: XML -*- -->
<!--
 This work is sponsored by the Department of the Air Force under Air Force
 Contract FA8721-05-C-0002. Opinions, interpretations, conclusions, and
 recommendations are those of the author and are not necessarily endorsed by
 the United States Government.
-->

<launch>
  <arg name="run_camera" default="true"/>

  <include file="$(find racecar)/launch/includes/joy_tsb_and_estop.launch.xml" />
  <include file="$(find racecar)/launch/includes/steering.launch.xml" />
  <include file="$(find racecar)/launch/includes/throttle.launch.xml" />

    <include file="$(find racecar)/launch/includes/sensors.launch.xml">
      <arg name="run_camera" value="false" />
    </include> 
  <include file="$(find racecar)/launch/mapping.launch" />

  <node pkg="racecar" type="VxEstimator" name="VxEstimator" output="screen">
	<remap from="/vx_filtered" to="/vx_filtered_from_optical_flow"/>
  </node>
  <node pkg="racecar" type="VxEstimatorFromSlam" name="VxEstimatorSlam" output="screen">
	<param name="frequency" value="50"/>
	<param name="learning_rate" value="0.3"/>
  </node>
  <node pkg="racecar" type="tbs2ts" name="tbs2ts" output="screen">
	<param name="frequency" value="100"/>
	<param name="max_brake" value="1"/>
	<param name="max_throttle" value="0.4"/>
	<param name="vx_thres" value="0.5"/>
  </node>

  <!-- <include file="$(find racecar)/launch/includes/sensors.launch.xml">
    <arg name="run_camera" value="$(arg run_camera)" />
  </include> -->
  
  <!-- <node pkg="racecar" type="laser_to_estop" name="laser_to_estop" output="screen"/> -->
  <node pkg="rosbag" type="record" name="bag_recorder"
	args="record -O /media/DATA/bags/test_telemetry.bag -a" />

</launch>
